src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -610,6 +618,82 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
         }
     }
 
+    /**
+     * @brief Checks and translates package information.
+     *
+     * This function searches for translation data in the Level 1 and Level 2 caches for a given package.
+     * If translation data is found, it populates the translations and returns them. If no translation data
+     * is found, it logs a debug message and returns an empty vector.
+     *
+     * @param package The package data to be checked and translated.
+     * @param osPlatform The operating system platform for which the package translation is required.
+     * @return A vector containing the translated package data.
+     */
+    std::vector<PackageData> checkAndTranslatePackage(const PackageData& package, const std::string& osPlatform)
+    {
+        std::vector<PackageData> vulnerabilityTranslations;
+        const auto cacheKey = osPlatform + "_" + package.vendor + "_" + package.name;
+
+        auto translatePackage = [&](const auto& translations)
+        {
+            for (const auto& translation : translations)
+            {
+                PackageData translatedPackage = package;
+                if (!translation.translatedProduct.empty())
+                {
+                    translatedPackage.name = translation.translatedProduct;
+                }
+                if (!translation.translatedVendor.empty())
+                {
+                    translatedPackage.vendor = translation.translatedVendor;
+                }
+                if (!translation.translatedVersion.empty())
+                {
+                    translatedPackage.version = translation.translatedVersion;
+                }
+                vulnerabilityTranslations.push_back(translatedPackage);
+            }
+        };
+
+        // Check Level 1 cache
+        if (m_translationFilter->count(cacheKey) == 0 && m_translationL1Cache->isHit(cacheKey))
```

Comment:
We should add a first if that checks the filter, if the translation is there, we return

Reply:
I think we're implicitly doing this, taking advantage of the `short-circuit` evaluation (do not evaluate the second operand if the result is known after evaluating the first)

Reply:
Yes, but at the same time we are losing readability (we could add a specific log for the first check)